common/src/main/java/com/graphicsfuzz/common/util/UpgradeShadingLanguageVersion.java

```diff
@@ -220,53 +299,61 @@ public void visitVariablesDeclaration(VariablesDeclaration variablesDeclaration)
 
   @Override
   public void visitFunctionCallExpr(FunctionCallExpr functionCallExpr) {
-    /*
-     Replace legacy function names with current ones.
-     Originally texture fetch functions included information about the kind of sampler
-     to be used, but later on these were simplified and the remaining texture fetch
-     functions work with various samplers.
-    */
-    switch (functionCallExpr.getCallee()) {
-      case "shadow1D":
-      case "shadow2D":
-      case "texture1D":
-      case "texture2D":
-      case "texture3D":
-      case "textureCube":
-        functionCallExpr.setCallee("texture");
-        break;
-      case "shadow1DProj":
-      case "shadow2DProj":
-      case "texture1DProj":
-      case "texture2DProj":
-      case "texture3DProj":
-        functionCallExpr.setCallee("textureProj");
-        break;
-      case "shadow1DLod":
-      case "shadow2DLod":
-      case "texture1DLod":
-      case "texture2DLod":
-      case "texture3DLod":
-      case "textureCubeLod":
-        functionCallExpr.setCallee("textureLod");
-        break;
-      case "shadow1DProjLod":
-      case "shadow2DProjLod":
-      case "texture1DProjLod":
-      case "texture2DProjLod":
-      case "texture3DProjLod":
-        functionCallExpr.setCallee("textureProjLod");
-        break;
-      default:
-        break;
+    // If a new name is in the map, rename the function
+    if (functionRename.containsKey(functionCallExpr.getCallee())) {
+      functionCallExpr.setCallee(functionRename.get(functionCallExpr.getCallee()));
     }
     for (Expr e : functionCallExpr.getArgs()) {
       visit(e);
     }
   }
 
+  @Override
+  public void visitFunctionPrototype(FunctionPrototype functionPrototype) {
+
```

Comment:
Similar comment here for the formatting of TODO.

Reply:
Done.